packages/labs/eleventy-plugin-lit/README.md

```diff
@@ -1,68 +1,396 @@
+displayed in some browsers.
+
+## Hydration
+
+_Hydration_ is the process where statically pre-rendered components are upgraded
+to their JavaScript implementations, becoming responsive and interactive.
+
+Lit components can automatically hydrate themselves when they detect that a
+Shadow Root has already been attached, as long as Lit's _experimental hydrate
+support_ module has been installed by importing
+[`lit/experimental-hydrate-support.js`](https://github.com/lit/lit/blob/main/packages/lit-element/src/experimental-hydrate-support.ts).
+
+> ⏱️ The Lit hydration support module **must be loaded before Lit or any
+> components that depend on Lit are imported**, because it modifies the initial
+> startup behavior of the `lit-element.js` module and the `LitElement` class.
+
+## Bootup
+
+It is important to preserve some constraints when designing a boot-up strategy
+for pages that use pre-rendered Lit components. In particular:
+
+- The Declarative Shadow DOM ponyfill must wait until all HTML has been parsed.
+- Lit and Lit component definition modules must wait until the experimental Lit
+  hydration support module has loaded.
+- Lit component definition modules must wait until the Declarative Shadow DOM
+  ponyfill to have been invoked (if it was needed for the browser).
+
+In the following diagram, each `->` edge represents a timing sequence
+constraint:
+
+```
+parse    load       install lit
+ HTML  ponyfill   hydration support
+  |       |        |
+  v       v        v
+ run ponyfill    load lit
+          |        |
+          v        v
+        load component
+         definitions
+```
+
+### Example bootup strategy
+
+> 🚧 Note: The pattern described here will only work in modern browsers such as
+> Firefox, Chrome, Edge, and Safari. IE11 is also supported, but will require a
+> different pattern that is not yet documented here. Documenting this pattern is
+> on the [roadmap](#roadmap). Follow
+> [#2494](https://github.com/lit/lit/issues/2486) for progress and discussion.
+> 🚧
+
+The following demonstrates an example strategy for booting up a page that
+contains pre-rendered Lit components with Eleventy.
+
+Typically in Eleventy your content is written in Markdown files which delegate
+the outer HTML shell to a `layout`. For example `hello.md` could delegate to the
+`default.html` layout like this:
+
+```markdown
+---
+layout: default.html
+---
+
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
 ```
 
-## Notes
+The file `_includes/default.html` would then contain the following:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <!-- As an optimization, immediately begin fetching the JavaScript modules
+        that we know for sure we'll eventually need. It's important we don't
+        execute them yet, though. -->
+    <link
+      rel="modulepreload"
+      href="/node_modules/lit/experimental-hydrate-support.js"
+    />
+    <link rel="modulepreload" href="/_js/component1.js" />
```

Comment:
We should add helpers for this. Could we add an issue and a note?

Reply:
Done (https://github.com/lit/lit/issues/2487 and https://github.com/lit/lit/issues/2490). Added a note.